### PR TITLE
[Speed] Adds Speculation docs

### PR DESCRIPTION
**Pull request by angelampcosta**
### Summary

Adds Speculation docs

### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+<FeatureTable id="speed.speculation" />
+
+---
+
+## Requirements
+
+Speculation works under the following conditions:
+- The Speculation feature is enabled in Cloudflare.
+- The browser of the web page visitor is using a Chromium-based browser v121 or later.
+- The web page requested by the prefetch is eligible for cache.
```
Comment (ngayerie) [inline src/content/docs/speed/optimization/content/speculation.mdx:27]: Should we point here to the documentation on how to cache HTML?
https://developers.cloudflare.com/cache/how-to/cache-rules/examples/cache-everything/

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+<FeatureTable id="speed.speculation" />
+
+---
+
+## Requirements
+
+Speculation works under the following conditions:
+- The Speculation feature is enabled in Cloudflare.
+- The browser of the web page visitor is using a Chromium-based browser v121 or later.
```
Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx]:
```suggestion
- The browser of the web page visitor is using a Chromium-based browser version 121 or later.
```

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    "eagerness": "conservative"
+  }]
+}
+```
+
+This configuration instructs the browser to initiate prefetch requests for future navigations. These prefetch requests will include the `sec-purpose: prefetch` HTTP request header. Prefetches that are not successful will respond with a `503` status code. Prefetches that are successful will respond with a `200` status code.
+
+## Test Speculation
+
+To test that Speculation is enabled, you can check that your HTTP response headers for your web pages include the `Speculation-Rules` header. However, note that during the Beta phase of Speculation, this behavior might not be 100% consistent.
```
Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx]: You mention `beta phase` here. Does this mean that this needs a beta tag in the title of the page?

Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx]:
```suggestion
To test that Speculation is enabled, you can check that your HTTP response headers for your web pages include the `Speculation-Rules` header. However, note that during the beta phase of Speculation, this behavior might not be 100% consistent.
```

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+To test whether your browser is making prefetch requests, open the **Network** tab in Chrome DevTools. Then, mouse-down on a link on a webpage with Speculation enabled. This action should initiate a prefetch request, which will be  visible in the **Network** tab. However, note that there are several reasons why the browser might choose not to initiate a prefetch. Refer to the [Chrome Limits guide](https://developer.chrome.com/docs/web-platform/prerender-pages#chrome-limits) for more details. For more general information about debugging Speculation, refer to the [Chrome Speculation Debugging guide](https://developer.chrome.com/docs/devtools/application/debugging-speculation-rules).
```
Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx:67]:
```suggestion
To test whether your browser is making prefetch requests, open the **Network** tab in Chrome DevTools. Then, select a link on a webpage with Speculation enabled. This action should initiate a prefetch request, which will be  visible in the **Network** tab. However, note that there are several reasons why the browser might choose not to initiate a prefetch. Refer to the [Chrome Limits guide](https://developer.chrome.com/docs/web-platform/prerender-pages#chrome-limits) for more details. For more general information about debugging Speculation, refer to the [Chrome Speculation Debugging guide](https://developer.chrome.com/docs/devtools/application/debugging-speculation-rules).
```

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+## RUM integration
+
+Speculation is designed to integrate with Web Analytics & Real User Measurements (RUM). This integration allows you to understand the web performance implications of Speculation within the Web Analytics UI in Cloudflare’s Dashboard.
```
Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx]:
```suggestion
Speculation is designed to integrate with Web Analytics & Real User Measurements (RUM). This integration allows you to understand the web performance implications of Speculation within the Web Analytics interface in Cloudflare’s Dashboard.
```

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+Speculation is available in Cloudflare’s **Speed** tab of the dashboard and also in the API and Terraform.
+
+<Tabs syncKey="dashPlusAPI"> <TabItem label="Dashboard">
+
+To enable **Prefetch URLs** in the dashboard:
+
+1. Log in to your [Cloudflare account](https://dash.cloudflare.com), select your account and go to a specific domain.
+2. Go to **Speed** > **Optimization** > **Content Optimization**.
+3. For **Speculation**, switch the toggle to **On**.
```
Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx]:
```suggestion
3. Enable **Speculation**.
```

File: src/content/docs/speed/optimization/content/speculation.mdx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+curl --request PATCH \
+"https://api.cloudflare.com/client/v4/zones/{zone_id}/settings/speculation" \
+--header "Authorization: Bearer <API_TOKEN>" \
+--header "Content-Type: application/json" \
+--data '{
+  "value":"on"
+  }
+```
+
+To disable Speculation, set value to "off".
```
Comment (marciocloudflare) [inline src/content/docs/speed/optimization/content/speculation.mdx]:
```suggestion
To disable Speculation, set `value:` to `"off"`.
```

**Review by marciocloudflare — APPROVED** [review]

**Pull request merged**